core/modules/layout/plugins/context/layout_context.inc

```diff
@@ -173,6 +181,35 @@ abstract class LayoutContext {
   function isA($plugin) {
     return $plugin === $this->plugin || in_array($plugin, $this->aliases);
   }
+
+  /**
+   * Settings form for configuring this context item.
+   */
+  function form(&$form, &$form_state) {
+  }
+
+  /**
+   * Validate handler for the settings form for this context item.
+   */
+  function formValidate($form, &$form_state) {
+
+  }
+
+  /**
+   * Submit handler for the settings form for this context item.
+   */
+  function formSubmit($form, &$form_state) {
+    $excluded = backdrop_map_assoc(array('cancel', 'submit', 'form_build_id', 'form_token', 'form_id', 'op', 'add', 'context'));
+    $settings = array_diff_key($form_state['values'], $excluded);
+    $this->settings = array_merge($this->settings, $settings);
```

Comment:
For a minute I thought putting the settings in a wrapper (e.g. `$form_state['values']['context_settings']`) would be better than excluding unwanted keys, but I see this is the same approach used for visibility conditions, so +1 for consistency.